### DISCUSSION TITLE
Can not add child element <Bundle> to <Wix>

### DISCUSSION THREAD

**Discussion by thomasengdaghl** [wixtoolset/issues#8815] [Questions]
I can not add the child element <Bundle> to <Wix>
Like:
  
 <Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" 

	<Package 			
		 Name="$(var.ProductName)"					
		 Manufacturer="$(var.Manufacturer)"
		 Version="14.0.0.4"
		 UpgradeCode="$(var.UpgradeCode)" 
		 ProductCode="*"  
        </Package>

       <Bundle Name="MyCompany MySoftware"  etc>
             <Chain>
                    <ExePackage  etc</ExePackage>
             </Chain>
       </Bundle>>
</Wix>

Under the Bundle keyword i have "squigle" lines

When compiling, I receive :
The element 'Wix' in namespace 'http://wixtoolset.org/schemas/v4/wxs' has invalid child element 'Bundle'. List of possible elements expected: 'Fragment' in namespace 'http://wixtoolset.org/schemas/v4/wxs'.	
Installer	C:\Users\notne\source\Workspaces\Picsara\2024-Branch\Installer\Package.wxs	120		

I'm using Visual Studio.
I have installed HeatWave:
I have tried on two different computers, same result.

In my project file i have:

<Project Sdk="WixToolset.Sdk/5.0.2">
  <PropertyGroup Label="Globals">
  
The documentation clearly says Bundle should go under <Wix>

I'm trying to add an external installer.exe to my installer project, have spent two days on this now and getting quite frustrated!!!
Please help

/Thomas

**Comment by robmen** [comment]
The [documentation](https://wixtoolset.org/docs/schema/wxs/wix/) also says this:

> This is the top-level container element for every wxs file. Among the possible children, the Bundle, Package, Module, Patch elements are analogous to the main function in a C program. **There can only be one of these present** when linking occurs. Package compiles into an MSI file, Module compiles into an MSM file, Patch compiles into an MSP file. The Fragment element is an atomic unit which ultimately links into either a Package, Module. The Fragment can either be completely included or excluded during linking.

_emphasis mine_

So, remove the `Package` element.

**Comment by thomasengdaghl** [comment]
Thank you Rob for quick response.

It's a bummer this was not easier spoted in the documentation, as I have spent 2 days figuring out why it does not work. and trying all kinds of trick to solve my root problem. Anyway I guess the problem lies in what is possible to do with a msi :-)

Anyway, I still have the problem how to add an external installer (when in the form of a msi or exe)
As it looks I need to create a new Bundle Project, and add both my own project and the external setup.exe to this.

 Anyway,  I realy appreciated your quick repsonse.
 /Thomas
